services/itunes/itunes.service.js

```diff
@@ -1,79 +1,61 @@
 'use strict'
 
-const LegacyService = require('../legacy-service')
-const { makeBadgeData: getBadgeData } = require('../../lib/badge-data')
-const { addv: versionText } = require('../../lib/text-formatters')
-const { version: versionColor } = require('../../lib/color-formatters')
+const Joi = require('joi')
+const BaseJsonService = require('../base-json')
+const { renderVersionBadge } = require('../../lib/version')
+const { NotFound } = require('../errors')
+const { nonNegativeInteger } = require('../validators')
 
-// This legacy service should be rewritten to use e.g. BaseJsonService.
-//
-// Tips for rewriting:
-// https://github.com/badges/shields/blob/master/doc/rewriting-services.md
-//
-// Do not base new services on this code.
-module.exports = class Itunes extends LegacyService {
+const schema = Joi.object({
+  resultCount: nonNegativeInteger,
+  results: Joi.array()
+    .items(Joi.object({ version: Joi.string().required() }))
+    .min(0),
+}).required()
+
+module.exports = class DubVersion extends BaseJsonService {
```

Comment:
guessing the class name here (`DubVersion`) is a mistake?

Reply:
whoops ;)